readFile.js

```diff
@@ -4,26 +4,32 @@ const PROG_BAR = '=';
 const BAR_LENGTH = 30;
 const HANG_PERCENT = 0.25;
 
-
-function readFile(fileToContent) {
-  let finalString;
-  const totalSize = fileSystem.statSync(process.argv[2]).size;
+async function readFile(filesToUpload) {
+  let promiseList = [];
+  const totalSize = filesToUpload.reduce((size, fileName) => (
+    size + fileSystem.statSync(fileName.content).size
+  ), 0);
   console.log(`Total size: ${totalSize}\nUpload progress:`);
-  const readStream = fileSystem.createReadStream(fileToContent);
-  return new Promise((resolve, reject) => {
-    try {
-      readStream.on('data', (chunk) => {
-        finalString += chunk;
-        const prog = ((chunk.length / totalSize) * BAR_LENGTH);
-        process.stdout.write(`${PROG_BAR.repeat(Math.round(prog - HANG_PERCENT))}`);
-      });
-      readStream.on('close', () => {
-        resolve(finalString);
-      });
-    } catch (error) {
-      reject(error);
-    }
+  promiseList = filesToUpload.map((file) => {
+    let finalContents = '';
+    return new Promise((resolve, reject) => {
+      try {
+        const readStream = fileSystem.createReadStream(file.content);
+        readStream.on('data', (chunk) => {
+          finalContents += chunk;
+          const prog = ((chunk.length / totalSize) * BAR_LENGTH);
+          process.stdout.write(`${PROG_BAR.repeat(Math.round(prog - HANG_PERCENT))}`);
+        });
+        readStream.on('close', () => {
+          resolve({ [file.name]: { content: finalContents } });
+        });
+      } catch (error) {
+        reject(error);
+      }
+    });
   });
+  return Promise.all(promiseList).then(response => response);
```

Comment:
`Promise.all(promiseList).then(response => response)` is exactly the same as `Promise.all(promiseList)`

Reply:
done